ivy/functional/ivy/gradients.py

```diff
@@ -37,17 +37,42 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 # Gradient Mode #
 
 # noinspection PyShadowingNames
-def with_grads(with_grads=None):
-    """Summary.
+def with_grads(with_grads: bool = None) -> bool:
+    """
+    Enter a nested code space where gradients are computed. This method
+    adds the with_grads component to the global list with_grads_stack
 
     Parameters
     ----------
     with_grads
-         (Default value = None)
+        Boolean value denoting whether the current code block has gradient
+        computation enabled or not.
+        'True' or 'False' or 'None' (Default value = None)
 
     Returns
     -------
     ret
+        If with_grads is boolean, it returns the boolean value representing
+        if gradient computation is enabled or not.
+        If with_grads is None, it returns the last element in the with_grads_stack
+        representing the parent of the current nested code block. If with_grads_stack
+        is empty, it returns True by default.
+        If with_grads is neither None nor boolean, it will raise an AssertionError
+
+    Examples
+    --------
+    Can be used by calling with ivy.with_grads(with_grads=...)
```

Comment:
Hi @sampreet-arthi , the examples are looking good!
You can delete line 64 as this line isn't necessary.

Reply:
Okay sure, is there anything I should do regarding the failing checks?